wdl-analysis/src/analyzer.rs

```diff
@@ -242,7 +286,7 @@ impl SourceEdit {
         self.range.start..self.range.end
     }
 
-    /// Applies the edit to the given string.
+    /// Applies the edit to the given string if it's in range.
     pub(crate) fn apply(&self, source: &mut String, lines: &LineIndex) {
```

Comment:
Attempting to apply an edit that isn't in range appears like something that should
return a `Result`. When that happens, is that typically going to be an issue on the
_client_ side of things (assuming no bugs in our own code, which we'd clearly want to flag)?

Reply:
The edit comes from a notification in the LSP, so there's no response we send the client to let them know the range data they gave us was invalid.

The best we could do is fail a subsequent diagnostics request, which is when we're applying the edits anyway.

Reply:
That's what I was afraid of.

In that case, since it would really be the client and not whatever code wrapped our analyzer (`wdl-lsp` or otherwise), I think we should simply emit a warning saying that the edit will be ignored because it was out of range. That way, the indication will be clear to anyone who wants to investigate.

Reply:
I decided to thread `Result` back through on the analysis request, so if the client send us an invalid edit, requests the analyze the file will fail.